cms/static/js/views/license.js

```diff
@@ -0,0 +1,136 @@
+        events: {
+            "click ul.license-types li button" : "onLicenseClick",
+            "click ul.license-options li": "onOptionClick"
+        },
+
+        initialize: function(options) {
+            this.licenseInfo = options.licenseInfo || defaultLicenseInfo;
+            this.showPreview = !!options.showPreview; // coerce to boolean
+            this.template = this.loadTemplate("license-selector");
+
+            // Rerender when the model changes
+            this.listenTo(this.model, 'change', this.render);
+            this.render();
+        },
+
+        getDefaultOptionsForLicenseType: function(licenseType) {
+            if (!this.licenseInfo[licenseType]) {
+                // custom license type, no options
+                return {};
+            }
+            if (!this.licenseInfo[licenseType].options) {
+                // defined license type without options
+                return {};
+            }
+            var defaults = {};
+            _.each(this.licenseInfo[licenseType].options, function(value, key) {
+                defaults[key] = value.default;
+            })
+            return defaults;
+        },
+
+        render: function() {
+            this.$el.html(this.template({
+                model: this.model.attributes,
+                licenseString: this.model.toString() || "",
+                licenseInfo: this.licenseInfo,
+                showPreview: this.showPreview,
+                previewButton: false,
+            }));
+            return this;
+        },
+
+        onLicenseClick: function(e) {
+            var $li = $(e.srcElement || e.target).closest('li');
+            var licenseType = $li.data("license");
+            this.model.set({
+                "type": licenseType,
+                "options": this.getDefaultOptionsForLicenseType(licenseType)
+            });
+        },
+
+        onOptionClick: function(e) {
+            var licenseType = this.model.get("type"),
+                licenseOptions = $.extend({}, this.model.get("options")),
+                $li = $(e.srcElement || e.target).closest('li');
+
+            var optionKey = $li.data("option")
+            var licenseInfo = this.licenseInfo[licenseType];
+            var optionInfo = licenseInfo.options[optionKey];
+            if (optionInfo.disabled) {
+                // we're done here
+                return;
+            }
+            var currentOptionValue = licenseOptions[optionKey];
+            if (optionInfo.type === "boolean") {
+                // toggle current value
+                currentOptionValue = !currentOptionValue;
+                licenseOptions[optionKey] = currentOptionValue;
+            }
+            // check for conflicts
+            if (currentOptionValue && optionInfo.conflictsWith &&
+                    _.any(optionInfo.conflictsWith, function (key) { return licenseOptions[key];})) {
+                // conflict! don't set new options
+                // need some feedback here
```

Comment:
TODO: Somewhere around here need to change these click events. Specifically if I click on something and there's a conflict wtih another, this behavior (from marco) should result: "hen an author clicks an unchecked Share Alike, it should uncheck No Derivatives and check Share Alike, and vice versa. Currently the checkboxes appear broken when you try to between them, and you must read the text to glean why this isn't behaving as a checkbox."